tools/run_tests/run_performance_tests.py

```diff
@@ -471,11 +540,26 @@ def finish_qps_workers(jobs):
     finally:
       # Consider qps workers that need to be killed as failures
       qps_workers_killed += finish_qps_workers(scenario.workers)
+      if perf_cmd and scenario_failures == 0:
```

Comment:
why do this in the finally block? 
1. you are generating the flamegraph only if there were no scenario failures anyway.
2. throwing in the finally block is not a good practice in general (in case there was and exception in the try block it eats it a throws a different one instead of it)

just run your code in the try block or after the entire try-finally.